src/components/Button.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { ButtonHTMLAttributes, MouseEvent, KeyboardEvent, ReactNode } from 'react';
+import { LoadingIndicator } from 'react-file-utils';
+
+export type ButtonProps = {
+  buttonStyle: 'info' | 'primary' | 'faded';
+  children: ReactNode;
+  disabled: boolean;
+  loading: boolean;
+  onClick: (event: MouseEvent<HTMLButtonElement>) => void;
+  onKeyPress: (event: KeyboardEvent<HTMLButtonElement>) => void;
+  type: ButtonHTMLAttributes<HTMLButtonElement>['type'];
+};
+
+export const Button = ({
+  buttonStyle = 'info',
+  disabled = false,
+  loading = false,
+  type = 'button',
+  onClick,
+  onKeyPress,
+  children,
+}: Partial<ButtonProps>) => (
```

Comment:
let's make the props optional in the type itself
```suggestion
}: ButtonProps) => (
```